network/endpoint.go

```diff
@@ -150,15 +150,31 @@ type apipaClient interface {
 	CreateHostNCApipaEndpoint(ctx context.Context, networkContainerID string) (string, error)
 }
 
+func FormatStructPointers[T any](slice []*T) string {
```

Comment:
`FormatPointerArrayToString` or `TransformPointerArrayToString`?

Reply:
what if array is not a pointer?

Reply:
if we try to pass in an array of structs instead of pointers, there will be a compilation error